src/wp-includes/blocks.php

```diff
@@ -1358,3 +1358,153 @@ function _wp_multiple_block_styles( $metadata ) {
+			set_query_var( 'cpage', $comment_args['paged'] );
+		}
+	}
+
+	return $comment_args;
+}
+
+/**
+ * Helper function that returns the proper pagination arrow html for
+ * `CommentsPaginationNext` and `CommentsPaginationPrevious` blocks based on the
+ * provided `paginationArrow` from `CommentsPagination` context.
+ *
+ * It's used in CommentsPaginationNext and CommentsPaginationPrevious blocks.
+ *
+ * @since 6.0.0
+ *
+ * @param WP_Block $block           Block instance.
+ * @param string   $pagination_type Type of the arrow we will be rendering.
+ *                                  Default 'next'. Accepts 'next' or 'previous'.
+ *
+ * @return string|null Returns the constructed WP_Query arguments.
+ */
+function get_comments_pagination_arrow( $block, $pagination_type = 'next' ) {
+	$arrow_map = array(
+		'none'    => '',
+		'arrow'   => array(
+			'next'     => '→',
+			'previous' => '←',
+		),
+		'chevron' => array(
+			'next'     => '»',
+			'previous' => '«',
+		),
+	);
+	if ( ! empty( $block->context['comments/paginationArrow'] ) && ! empty( $arrow_map[ $block->context['comments/paginationArrow'] ][ $pagination_type ] ) ) {
+		$arrow_attribute = $block->context['comments/paginationArrow'];
+		$arrow           = $arrow_map[ $block->context['comments/paginationArrow'] ][ $pagination_type ];
+		$arrow_classes   = "wp-block-comments-pagination-$pagination_type-arrow is-arrow-$arrow_attribute";
+		return "<span class='$arrow_classes'>$arrow</span>";
+	}
+	return null;
+}
+
+/**
+ * Workaround for getting discussion settings as block editor settings
+ * so any user can access to them without needing to be an admin.
+ *
+ * @since 6.0.0
+ *
+ * @param array $settings Default editor settings.
+ *
+ * @return array Filtered editor settings.
+ */
+function extend_block_editor_settings_with_discussion_settings( $settings ) {
+
+	$settings['__experimentalDiscussionSettings'] = array(
+		'commentOrder'        => get_option( 'comment_order' ),
+		'commentsPerPage'     => get_option( 'comments_per_page' ),
+		'defaultCommentsPage' => get_option( 'default_comments_page' ),
+		'pageComments'        => get_option( 'page_comments' ),
+		'threadComments'      => get_option( 'thread_comments' ),
+		'threadCommentsDepth' => get_option( 'thread_comments_depth' ),
+		'avatarURL'           => get_avatar_url(
+			'',
+			array(
+				'size'          => 96,
+				'force_default' => true,
+				'default'       => get_option( 'avatar_default' ),
+			)
+		),
+	);
+
+	return $settings;
+}
+add_filter( 'block_editor_settings_all', 'extend_block_editor_settings_with_discussion_settings' );
+
+/**
+ * Mark the `children` attr of comments as embeddable so they can be included in
+ * REST API responses without additional requests.
+ *
+ * @since 6.0.0
+ *
+ * @return void
+ */
+function gutenberg_rest_comment_set_children_as_embeddable() {
+	add_filter(
+		'rest_prepare_comment',
+		function ( $response ) {
+			$links = $response->get_links();
```

Comment:
This part also should land in the class that handles the REST API endpoint for comments. @spacedmonkey and @TimothyBJacobs will know best where to move it.

Reply:
Yeah, this needs to move into `WP_REST_Comments_Controller::prepare_links()`.